transports/noise/src/io/framed.rs

```diff
@@ -43,61 +39,92 @@ static_assertions::const_assert! {
     MAX_FRAME_LEN + EXTRA_ENCRYPT_SPACE <= MAX_NOISE_MSG_LEN
 }
 
-/// A `NoiseFramed` is a `Sink` and `Stream` for length-delimited
-/// Noise protocol messages.
-///
-/// `T` is the type of the underlying I/O resource and `S` the
-/// type of the Noise session state.
-pub(crate) struct NoiseFramed<T, S> {
-    io: T,
-    session: S,
-    read_state: ReadState,
-    write_state: WriteState,
-    read_buffer: Vec<u8>,
+/// Codec holds the noise session state `S` and acts as a medium for
+/// encoding and decoding length-delimited session messages.
+pub(crate) struct Codec<S> {
     write_buffer: Vec<u8>,
     decrypt_buffer: BytesMut,
+    session: S,
```

Comment:
Nit: Can we move this on top of the two fields to make this diff a little clearer?

Reply:
yh, can do!